drivers/serial/uart_rpi_pico.c

```diff
@@ -302,6 +311,9 @@ static const struct uart_driver_api uart_rpi_driver_api = {
 		.uart_dev = uart##idx,						\
 		.uart_regs = (uart_hw_t *)uart##idx,				\
 		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(idx),			\
+		.reset_ctrl = DEVICE_DT_GET(DT_INST_PHANDLE(idx, resets)),	\
+		.reset_offset = DT_INST_PHA_BY_IDX(idx, resets, 0, offset),	\
+		.reset_bit = DT_INST_PHA_BY_IDX(idx, resets, 0, regbit),	\
```

Comment:
A single concise macro, similar to what `PINCTRL_DT_INST_DEV_CONFIG_GET` is doing would be nice IMHO

Reply:
Okay, I will try to create some macros for this.

Reply:
> A single concise macro, similar to what `PINCTRL_DT_INST_DEV_CONFIG_GET` is doing would be nice IMHO

I did something, probably not perfect, but if you have another suggestions I will gladly receive them.